ovirt/resource_ovirt_vm.go

```diff
@@ -376,6 +382,71 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	vmBuilder.Cpu(cpu)
 
+	if storage_domain, _ := blockDevice.([]interface{})[0].(map[string]interface{})["storage_domain"]; storage_domain != "" && templateIDOK {
+
+		// Get the reference to the service that manages the storage domains
+		sdsService := conn.SystemService().StorageDomainsService()
+
+		// Find the storage domain we want to be used for virtual machine disks
+		sdsResp, err := sdsService.List().Search("name=" + storage_domain.(string)).Send()
+		if err != nil {
+			return fmt.Errorf("Failed to search storage domains, reason: %v", err)
+		}
+
+		tds, err := getTemplateDiskAttachments(templateID.(string), meta)
+		if err != nil {
+			return fmt.Errorf("Failed to get Template disks attachments: %v", err)
+		}
+
+		if sdSlice, ok := sdsResp.StorageDomains(); ok {
+			sd := sdSlice.Slice()[0]
+
+			for i, v := range tds {
+				diskIndex := i + 1
+				disk := v.MustDisk()
+				disk.SetStorageDomain(sd)
+
+				// Define basic disk aliases only if attribute alias defined
+				if alias, _ := blockDevice.([]interface{})[0].(map[string]interface{})["alias"]; alias != "" {
+					_, diskBotable := disk.Bootable()
+					switch diskBotable {
+					case true:
+						disk.SetAlias(alias.(string))
+					case false:
+						disk.SetAlias(fmt.Sprintf("%s_Disk%v", d.Get("name").(string), diskIndex))
+					}
+					diskattachment := ovirtsdk4.NewDiskAttachmentBuilder().
+						Disk(ovirtsdk4.NewDiskBuilder().
+							Id(disk.MustId()).
+							Format(ovirtsdk4.DISKFORMAT_COW).
```

Comment:
why always COW?

Reply:
COW - because I can't determine what disk format is inside current template

`diskFormat, err := disk.Format()` return empty string. Now exact value.
Therefore was used most common format "COW", that is the same as thin disk and allow snapshots.

If you know how to get template disk format, can you please update this code or give me some example?

Reply:
To the example was added note about disk destination format for users, who want to use this feature - https://github.com/oVirt/terraform-provider-ovirt/pull/216/commits/f67b88faf8248f1db29dc104a345d7a97b0068b0